pyanaconda/ui/common.py

```diff
@@ -683,3 +683,22 @@ def collectCategoriesAndSpokes(paths, klass):
     lifecycle.add_controller(klass.__name__, category_names)
 
     return ret
+
+
+def sort_categories(categories):
+    """Sort categories based on sort order & calls name.
+
+    While sort order is the primarily sorting key, we can't always
+    assure all categories have unique sort order values.
+    So make the class name part of the sorting key, to at least
+    make the resulting category sort order deterministic.
+
+    To do this we concatenate we create a tuple as the sorting key,
+    with the sort order integer being followed by the by the class
+    name. This way the categories will be sorted by the sort order
+    first and then by class name if sort order happens to be the same.
+
+    :param categories: list of categories
+    :type categories: SpokeCategory sub classes
+    """
+    return sorted(categories, key=lambda i: (i.get_sort_order(), i.__class__))
```

Comment:
It should be sorted by `__class__.__name__`.

Reply:
I couldn't figure out why your test is not failing, because classes are not comparable. The problem is that categories are not instances, but classes. Therefore `i` is a class and `i.__class__` is `type`. And `type` is comparable. Anyway, categories should be sorted by `__name__`. Please, make sure that your tests are testing this use case.

Reply:
Oops! Yeah, that's really wrong. :P I'll fix it up in a folowup PR.